src/Conf.ml

```diff
@@ -1839,14 +1834,13 @@ type action =
   | Check of [`Impl | `Intf | `Use_file] input list
 
 let kind_of fname =
-  match !kind with
-  | Some kind -> kind
-  | None -> (
-    match Filename.extension fname with
-    | ".ml" -> `Impl
-    | ".mli" -> `Intf
-    | ".mlt" -> `Use_file
-    | _ -> `Impl )
+  Option.value !kind
+    ~default:
+      ( match Filename.extension fname with
```

Comment:
This was computed only when needed so this is slightly less good.

Reply:
done